### DISCUSSION TITLE
Is there a trick to building on MacOS?

### DISCUSSION THREAD

**Discussion by Funkton** [OrcaSlicer/OrcaSlicer#251] [General]
Hey all,
I'm trying to build from source and it seems like there's some secret sauce I'm missing when attempting to build using the build_release_macos.sh file.

I've gotten through a lot of the cmake errors (xcode support broken?). I can get all the dependencies built, but when I run the full build it errors saying there are no files found doing the .app cp and rm commands.

Do I need to manually add existing .app files in a location? In the following command, there is no Release folder in any location
cp -pR ../src/Release/

Any help would be greatly appreciated. I don't have a lot of experience building software so it can be hard sometimes to work through a process that has no instructions.

The main reason I'm attempting this is to get a version of the SoftFever running on my Catalina machine. The last couple versions broke support and it says I have to run MacOS 12+. Stock BambuStudio runs fine on it.

**Comment by Elbau** [comment]
So I just did the build last night and it made me feel very rusty. I also don't have a lot of experience in this regard. I can't speak to your specific problem, but these are the steps that I took to build the app from source on my mac (M1 on MacOS 13.1).

- Clone the github repository to your local machine. I used Xcode > Source Control > Clone
- Install HomeBrew package manager
- follow the instructions to add HomeBrew to your command PATH
- Use homebrew to install CMake
- 'sudo xcode-select --reset'  -- I had to do this to fix the compiler associations 
- run the 'build_release_macos.sh' script

It sounds like the build is failing at assembling the .app container, so there must something going wonky upstream from that.

**Comment by SoftFever** [comment]
You will need to install these packages:
`brew install cmake git gettext`

**Reply by Funkton** [reply]
Yes I have all those installed. I've followed all the instructions in the compile guide.
I think the main issue is I'm trying to build on 10.15 and the dependency packages are not happy with that. I'm trying to build this at home on my M1 mac as well running Monterey. Also has many errors.

I'm getting errors with just about every package. Things like
`./boost/config/detail/posix_features.hpp:18:15: fatal error: 'unistd.h' file not found`
and
`/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/cassert:20:10: fatal error: 'assert.h' file not found`

```
...failed updating 67 targets...
...skipped 50 targets...
make[3]: *** [dep_Boost-prefix/src/dep_Boost-stamp/dep_Boost-build] Error 1
make[2]: *** [CMakeFiles/dep_Boost.dir/all] Error 2
make[1]: *** [CMakeFiles/deps.dir/rule] Error 2
```

My co-worker has been trying to build as well while getting other errors.

**Reply by Elbau** [reply]
Check to make sure you have the 'Xcode command line tools' installed and up-to-date.

**Reply by Funkton** [reply]
They are as up to date as they can be on MacOS 10.15 (xcode 12.4)

**Reply by SoftFever** [reply]
I’d suggest you follow @Elbau ‘s instructions. You will need to set up the tool aster installing Xcode

**Reply by Funkton** [reply]
Yes, I've done this on 3 machines now. Even attempted with vanilla BBS following their compile guide.
If the SoftFever fork was built with 10.15 set as the minimum I wouldn't need to build my own.

**Reply by hliebscher** [reply]
> Yes, I've done this on 3 machines now. Even attempted with vanilla BBS following their compile guide. If the SoftFever fork was built with 10.15 set as the minimum I wouldn't need to build my own.

pls delete the build directory in root and under /deps  then try again

**Reply by Ryszard-wpl** [reply]
half an hour of compiling and failed. Last warnings were mentioned that Deployment Target is 11.3 and I have 10.15 ... What to do. I used script for mac os after I made it executable.

**Reply by TonyK54** [reply]
> pls delete the build directory in root and under /deps then try again

compiling now. it would not compile until I deleted the build folders. Trying to add an Ender  5 Pro (2019) to see if I can use OrcaSlicer for all my needs.

**Comment by Manfredo1952** [comment]
I am having an iMac here with OSX 11.7.3, Intel - I had a problem getting the compilation to work correctly with error reported concerning openssl, not found and stuff.  Searched a bit in the internet and found a hint that I should amend the the cmake command to point to the directory in my files system here.
The respective command now is as follows (you may want to check your system for the same folder structure )
cmake -DOPENSSL_ROOT_DIR=/usr/local/Cellar/openssl/1.1.1lk -DOPENSSL_LIBRARIES=/usr/local/Cellar/openssl/1.1.1l/lib ..  

the entire .sh file, which worked for me, is as follows  - when copying the .app file to program folder first starts needs to be with CTRL-Click


[build_release_macos2.sh.zip](https://github.com/SoftFever/OrcaSlicer/files/10971521/build_release_macos2.sh.zip)
